underworld/mesh/_meshvariable.py

```diff
@@ -404,10 +413,24 @@ def save( self, filename, meshHandle=None ):
                                       shape=globalShape,
                                       dtype=self.data.dtype)
 
+            fact = 1.0
+            if units:
+                fact = dimensionalise(1.0, units=units).magnitude
+                if units == "degC":
+                    fact = dimensionalise(1.0, units=u.degK).magnitude
+                # Save unit type as attribute
+                h5f.attrs['units'] = str(units)
+
+            if time:
+                h5f.attrs["time"] = str(time)
+
             # write to the dset using the global node ids
             local = mesh.nodesLocal
             with dset.collective:
-                dset[mesh.data_nodegId[0:local],:] = self.data[0:local]
+                if units = "degC":
```

Comment:
@rbeucher why is this special case needed?  It this not directly supported in `pint`?

Reply:
no there is an ambiguity around temperature units that needs to be taken care of.

Reply:
👍👍